scripts/runBenchmarks.ts

```diff
@@ -0,0 +1,224 @@
+		if (i > 0) data += '\n\n';
+		data += `Result of ${results[i].endpoint}: \u2193 \n\n${results[i].stdout}`;
+	}
+
+	return data;
+};
+
+/**
+ * Write the results of the benchmarks to file. Per benchmark these will write over the existing benchmarks.
+ *
+ * @param results Results organized from stdout
+ */
+const writeResultsToFile = (results: IBenchResult[]) => {
+	const FILE_NAME = 'benchmarks.txt';
+	const formattedData = formatResults(results);
+	fs.writeFileSync(FILE_NAME, formattedData);
+};
+
+/**
+ * Launch a single benchmark. It returns the stdout from `<root>/benchmarks/util/util.lua`.
+ *
+ * @param endpoint Endpoint that reflects one of the keys from `benchmarkConfig`.
+ * @param wsUrl `wsUrl` to benchmark off of.
+ */
+const launchBenchmark = async (
+	endpoint: string,
+	wsUrl: string
+): Promise<string> => {
+	const { Failed } = StatusCode;
+	// Set the ws url env var
+	setWsUrl(wsUrl);
+
+	console.log('Launching Sidecar...');
+	const sidecarStart = await launchProcess('yarn', procs, defaultSasStartOpts);
+	if (sidecarStart.code === Failed) {
+		console.error('Sidecar failed to start');
+		killAll(procs);
+		process.exit(1);
+	}
+
+	// 2 second delay to allow sidecar to boot before we load it will queries.
+	await delay(2000);
+
+	// cd into benchmark
+	cdToDir(benchmarkConfig[endpoint].path);
+	// Run benchmark against the endpoint passed in
+	const bench = await launchProcess('sh', procs, {
+		proc: 'benchmark',
+		resolver: 'Benchmark finished',
+		args: ['init.sh'],
+	});
+	// cd back to root
+	cdToDir('/../../');
+
+	// Ensure the benches process was sucessful.
+	if (bench.code === Failed) {
+		console.log('ERROR code: 4 - Error running launchBenchmark()');
+		killAll(procs);
+		process.exit(4);
+	}
+	// Ensure before we exit the function that we exit all processes.
+	killAll(procs);
+
+	return bench.stdout;
+};
+
+const main = async (args: IBenchParseArgs) => {
+	const { Failed } = StatusCode;
+	const { log_level, endpoint, ws_url } = args;
+
+	if (log_level) setLogLevel(log_level);
+	setBenchTime(args.time);
+
+	console.log('Building Sidecar...');
+	const sidecarBuild = await launchProcess('yarn', procs, defaultSasBuildOpts);
+
+	if (sidecarBuild.code === Failed) {
+		console.log('Sidecar failed to build, exiting...');
+		killAll(procs);
+		process.exit(2);
+	}
+
+	const results: IBenchResult[] = [];
+	if (endpoint) {
+		const res = await launchBenchmark(endpoint, ws_url);
+		results.push({
+			endpoint,
+			stdout: res,
+		});
+	} else {
+		// Launch each benchmark for each endpoint
+		const endpoints = Object.keys(benchmarkConfig);
+		for (const endpoint of endpoints) {
+			const res = await launchBenchmark(endpoint, ws_url);
+			results.push({
+				endpoint,
+				stdout: res,
+			});
+		}
+	}
+
+	writeResultsToFile(results);
+};
+
+const parser = new ArgumentParser();
+
+parser.add_argument('--ws-url', {
+	required: false,
+	nargs: '?',
+	type: checkWsType,
+	default: 'ws://127.0.0.1:9944',
```

Comment:
what would happen if we specify one of our public endpoints here (f.e. "wss://rpc.polkadot.io")?

Reply:
It would use the public endpoint with no problem, but it is highly not recommended to use a public endpoint. 

Why?: Because the benchmarks will send hundreds of thousands of requests, and we don't want to flood it with storage queries.